src/modules/colorPicker/ColorPickerUI/Behaviors/ResizeBehavior.cs

```diff
@@ -16,12 +16,9 @@ public class ResizeBehavior : Behavior<FrameworkElement>
         private static void WidthPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
             var sender = ((ResizeBehavior)d).AssociatedObject;
-            var move = new DoubleAnimation(sender.Width, (double)e.NewValue, new Duration(TimeSpan.FromMilliseconds(150)), FillBehavior.Stop);
-            move.Completed += (s, e1) =>
-            {
-                sender.BeginAnimation(FrameworkElement.WidthProperty, null);
-                sender.Width = (double)e.NewValue;
-            };
+            sender.Width = (double)e.NewValue;
+
+            var move = new DoubleAnimation(sender.Width, (double)e.NewValue, new Duration(TimeSpan.FromMilliseconds(100)), FillBehavior.Stop);
```

Comment:
Stupid q, i read line 19-21 as you're setting the height to the new value then doing an animation to the same value?  32 the value is set so 21 the from and to are the same value.

Reply:
I didn't notice that, good catch!
Looks like this works because `sender.Width` isn't updated until after the function is already over, so `new DoubleAnimation` still reads the original value.
However, there's once exception, which is probably why I put it above the animation in the first place.
If `sender.Width` hasn't been set before (the window was just created), then `new DoubleAnimation` *will* get the value it was set to. This is actually desirable, because otherwise there would be an animation from 0 to 50 the first time the zoom window is created, then never again.
Considering this, I have two ideas on how to make this behavior more robust:
1. Set the new Width/Height below the `new DoubleAnimation`, and also set the desired Width/Height to 0 when the zoom window is closed. Doing this will mean there's the animation from 0 to 50 every time a zoom is started.
2. Set the new Width/Height below the `new DoubleAnimation`, but first check if the Width/Height is 0. If it is, simply skip the animation and directly set the new values.

Approach 1:

https://user-images.githubusercontent.com/5243039/117236042-40649780-ae6b-11eb-93b7-d89ac8e85f94.mp4

Approach 2:

https://user-images.githubusercontent.com/5243039/117236073-540ffe00-ae6b-11eb-9f8c-14177c8c4a52.mp4

Reply:
i think the bottom one looks a tad smoother.  My suggestion is comment why you're doing what you're doing as that line raised an eyebrow